Compile/src/Producers/LeptonSFProducer.cc

```diff
@@ -300,64 +315,139 @@ std::string LeptonRecoSFProducer::GetProducerId() const { return "LeptonRecoSFPr
+        if (eta < 0.9) {
+            sf = 0.9996742562806361;
+            stat = 7.650191371261136e-05;
+            syst = 0.0006514874387277825;
+        }
+        else if (eta < 1.2) {
+            sf = 0.9997813602035737;
+            stat = 0.00014496238686164667;
+            syst = 0.0004372795928526685;
+        }
+        else if (eta < 2.1) {
+            sf = 0.9994674742459532;
+            stat = 7.739510750489317e-05;
+            syst = 0.0010650515080936618;
+        }
+        else if (eta < 2.4) {
+            sf = 0.9993566412630517;
+            stat = 0.00022835790507860388;
+            syst = 0.0011810962222705494;
+        }
+        else {
+            LOG(ERROR) << "No RecoSF available for this p and eta region. Using 1+/-0";
+        }
+    }
+    else if (m_year == "2016") {
+        //https://gitlab.cern.ch/cms-muonPOG/muonefficiencies/-/blob/master/Run2/UL/2016_postVFP/NUM_TrackerMuons_DEN_genTracks_Z_abseta_pt.json
+        if (eta < 0.9) {
+            sf = 1.0000406419782646;
+            stat = 0.00010260291858070426;
+            syst = 0.0014366927652431664;
+        }
+        else if (eta < 1.2) {
+            sf = 0.9997959311146515;
+            stat = 0.00019912837537507789;
+            syst = 0.0010917857343065423;
+        }
+        else if (eta < 2.1) {
+            sf = 0.9994928400570587;
+            stat = 0.00012513847429973846;
+            syst = 0.0014814654032937547;
         }
+        else if (eta < 2.4) {
+            sf = 0.9990728619505579;
+            stat = 0.0002754474704705526;
+            syst = 0.0017364778744567663;
+        }
+        else {
+            LOG(ERROR) << "No RecoSF available for this p and eta region. Using 1+/-0";
+        }
+    }
+    else if (m_year == "2016APV") {
+        //https://gitlab.cern.ch/cms-muonPOG/muonefficiencies/-/blob/master/Run2/UL/2016_preVFP/NUM_TrackerMuons_DEN_genTracks_Z_abseta_pt.json
+        if (eta < 0.9) {
+            sf = 0.9998229551300333;
+            stat = 0.0001538802103231026;
+            syst = 0.0003540897399334497;
+        }
+        else if (eta < 1.2) {
+            sf = 1.0001593416915515;
+            stat = 0.00019861903120026457;
+            syst = 0.00031024592139106355;
+        }
+        else if (eta < 2.1) {
+            sf = 0.9998936144006075;
+            stat = 0.00012188589514012365;
+            syst = 0.00021277119878493345;
+        }
+        else if (eta < 2.4) {
+            sf = 0.9990268820042745;
+            stat = 0.00027638902644996395;
+            syst = 0.0019462359914510508;
+        }
+        else {
+            LOG(ERROR) << "No RecoSF available for this p and eta region. Using 1+/-0";
+        }
+    } else if (m_year != "none") {
+    LOG(FATAL) << "No high pt RecoSF available for year " << m_year;
     }
+    float err = sqrt(pow(stat, 2.0) + pow(syst, 2.0));
```

Comment:
Are you sure this is valid?

Reply:
Wouldn't it be corret to get the influences of statistical and systematical sources seperately and combine the resulting differences of each source on the histogram quadratically?

Reply:
Looking at the Root File of other SFs, the uncertainty of a bin is always sqrt(stat**2 + syst**2)
E.g. https://gitlab.cern.ch/cms-muonPOG/muonefficiencies/-/blob/master/Run2/UL/2018/2018_trigger/Efficiencies_muon_generalTracks_Z_Run2018_UL_SingleMuonTriggers.root
```
root [10] NUM_IsoMu24_DEN_CutBasedIdTight_and_PFIsoTight_abseta_pt->GetBinError(2,2)
(double) 0.00047113752
root [11] NUM_IsoMu24_DEN_CutBasedIdTight_and_PFIsoTight_abseta_pt_stat->GetBinError(2,2)
(double) 0.00032228204
root [12] NUM_IsoMu24_DEN_CutBasedIdTight_and_PFIsoTight_abseta_pt_syst->GetBinError(2,2)
(double) 0.00034366387
```
```
>>> import numpy as np
>>> stat = 0.00032228204
>>> syst = 0.00034366387
>>> np.sqrt(stat**2 + syst**2)
0.00047113752647177075
```

Reply:
Okay, let's keep it like it is, then. For future studies, however, it might be beneficial to have them available as separate SFs, like you have it also for the other types, e.g. PUJetID.